### PR TITLE
Add is-selected to /sites on initial open preview pane

### PR DESCRIPTION
**Pull request by candy02058912**
Fixes https://github.com/Automattic/dotcom-forge/issues/6782

## Proposed Changes

This PR does direct DOM manipulation to add the `is-selected` class.

**Before**
See video in https://github.com/Automattic/dotcom-forge/issues/6782

**After**
See video in https://github.com/Automattic/dotcom-forge/issues/6782#issuecomment-2082467258

This is to prevent leakage of internal P2 URLs.

### Notes

- This is a very hacky proposal as it isn't supported in the `@wordpress/dataviews` package [located here](https://github.com/WordPress/gutenberg/blob/cf7e1ea241a8d872bb830b89cab9e65f2409b2fb/packages/dataviews/README.md).
- Even if we get a fix into the package mentioned above, our package.json is locked to a specific version of the `@wordpress/dataviews`, probably due to upgrading it might cause issues (https://github.com/Automattic/wp-calypso/pull/87956)

### Know Issues
Will need to wait until https://github.com/Automattic/dotcom-forge/issues/6803 is fixed for pagination-related issues.

## Testing Instructions

* Go to `/sites?flags=layout/dotcom-nav-redesign-v2`
* Click on a site
* See if the site is highlighted in the sidebar of the flyout panel

Try to break it.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by davemart-in** [comment]
@candy02058912 it looks like something in here might be breaking the transition effect. I'll ping you in the issue with a few screenshots.

**Comment by candy02058912** [comment]
Closing this PR as per the comment mentioned https://github.com/Automattic/dotcom-forge/issues/6782#issuecomment-2084257168

TL;DR, this seems possible to be fixed with a `@wordpress/dataviews` package update that we will be aiming for in the near future. https://github.com/Automattic/wp-calypso/pull/89739
